ui/src/components/FDashboard.tsx

```diff
@@ -0,0 +1,252 @@
+    data.subjectname.toLowerCase().includes(searchQuery.toLowerCase())
+  
+  // || contact.email.toString().includes(searchQuery.toLowerCase())
+    );
+
+    const search = (e :  React.ChangeEvent<HTMLInputElement>) => {
+      setSearchQuery(e.target.value)
+      setCurrentPage(0)
+    }
+    
+    const startIndex = currentPage * itemsPerPage;
+    const endIndex = startIndex + itemsPerPage;
+    const paginatedData = filteredData.slice(startIndex, endIndex);
+    
+    const handlePageChange = (selectedPage: { selected: number }) => {
+      setCurrentPage(selectedPage.selected);
+    };
+     
+    const totalPages = Math.ceil(filteredData.length / itemsPerPage);
+
+  return (
+    <div className="mt-1">
+      
+      {Editbool  || AddModal || manual != false ? (
+        <Modal body={
+          <>
+          {manual != false ? <Manual aid = {manual.id}/> : null}
+          {AddModal ? <AttendAddEdit handle={()=>setAddModal(false)}/> : null}
+          {Editbool ? <AttendAddEdit initialData={EditModal} handle={()=>seteditbool(false)}/> : null} 
+        </>
+      }
+      header = {
+        {title : AddModal ? 'Add Attendance Request' : Editbool ? 'Edit Attendance Request' : manual != false ?`${manual.date} - ${manual.classname} - ${manual.subjectname}` : '',
+        closehandle : AddModal ? ()=>{setAddModal(false)} : Editbool ? ()=>{seteditbool(false)} : ()=>{setManual(false)}}
+      }
+      /> 
+      ) : null}
+      
+       <div className="flex justify-evenly items-center mb-4">
+        <label htmlFor="search" className="block ml-6 text-sm font-medium text-gray-700">
+          Search by Subject name
+        </label>
+        <input
+          type="text"
+          id="search"
+          placeholder="Search..."
+          value={searchQuery}
+          onChange={(e) => search(e)}
+          className="mt-1 p-2  ml-5 w-96 border rounded-md focus:ring focus:ring-blue-200"
+        />
+        
+          <button className="bg-Aqua-300 text-Black font-medium m-1 py-2 px-4 rounded-md hover:font-bold ml-2"
+            onClick={() => setAddModal(true)}>
+            ADD
+          </button>
+          
+         
+      </div>
+      {filteredData.length != 0 ? (
+        <>
+        
+        <table className="w-11/12 mx-auto border-rounded border border-gray-300 shadow-md">
+        <thead>
+          <tr className="bg-Aqua-300 text-Black shadow-md ">
+          <th className="p-3 text-center">Date</th> 
+          <th className="p-3 text-center">Class Name</th>
+            <th className="p-3 text-center">Subject Name</th>
+            <th className="p-3 text-center">Periods</th>
+            <th className="p-3 text-center">Type</th>
+            <th className="p-3 text-center">Status</th>
+            <th className="p-3 text-center">Created</th>
+            <th className="p-3 text-center">Updated</th>
+            <th className="p-3 text-center">Actions</th>
+            <th className="p-3 text-center">  </th>
```

Comment:
In general, We won't be using text directly lik this. 
We'll use i18n. try it once. So that you guys will know how it works.